feat(circleci): Add support for CircleCi

karlderkaefer opened the pull request:
Adding support to generate pipelines for CircleCi

fixes #1717 
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Chriscbr commented:
Hey @karlderkaefer - thanks for taking initiative on this! This is definitely a feature we want in projen -- but it might be a while before we're able to get to review this. I appreciate your patience 🙏

Chriscbr commented on src/circleci/circleci.ts:
```diff
@@ -0,0 +1,222 @@
+  /**
+   * The yaml file for the Circleci pipeline
+   * */
+  public readonly file: YamlFile | undefined;
```
See other comment
```suggestion
  public readonly file: YamlFile;
```

Chriscbr commented on src/circleci/circleci.ts:
```diff
@@ -0,0 +1,222 @@
+   */
+  private readonly jobs: Job[];
+
+  /**
+   * create a Circleci Component within project
+   * @param project
+   * @param options
+   */
```
No need to have a docstring for the constructor
```suggestion
```

Chriscbr commented on test/cirlceci/circleci.test.ts:
```diff
@@ -0,0 +1,215 @@
+  expect(circleci).toContain("workflow1");
+  expect(circleci).toContain("world:2.0");
+  expect(circleci).not.toContain("world:3.0");
+  console.log(circleci);
```
```suggestion
```

Chriscbr commented on test/cirlceci/circleci.test.ts:
```diff
@@ -0,0 +1,215 @@
+  expect(circleci).toMatchSnapshot();
+  expect(circleci).toContain("customName2");
+  expect(circleci).toContain("context3");
+  console.log(circleci);
```
```suggestion
```

karlderkaefer commented on src/circleci/circleci.ts:
```diff
@@ -0,0 +1,222 @@
+    this.workflows = options.workflows ?? [];
+    this.jobs = options.jobs ?? [];
+    const circleCiEnabled = options.enabled || true;
+    if (circleCiEnabled) {
```
fixed in b882b4f97a84a417ed1da2a17e52f5dd46432840

karlderkaefer commented:
@Chriscbr thanks for your review and time, I have addressed all suggestions. I got one question to commit messages, I did not follow the conventional message on every commit, should I rebase and rewrite commit messages, or is that not so important since PRs getting squash merged anyway?
